client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala

```diff
@@ -98,7 +98,8 @@ object FollowRedirect {
         case Right((resp, dispose)) =>
           (methodForRedirect(req, resp), resp.headers.get(Location)) match {
             case (Some(method), Some(loc)) if redirects < maxRedirects =>
-              val nextReq = nextRequest(req, loc.uri, method, resp.cookies)
+              // Dotty needs the asInstanceOf help.
+              val nextReq = nextRequest(req, (loc.asInstanceOf[Location]).uri, method, resp.cookies)
```

Comment:
This is where the `HeaderT` projection starts to raise its ugly head.

Reply:
indeed. but you can avoid that by assigning 
`resp.headers.get(Location)` to a val

its still not good, but avoids asInstanceOf

Reply:
Oh, nice. I like that better.